xonsh/completers/bash_completion.py

```diff
@@ -432,6 +432,16 @@ def bash_completions(
     # to be incorrectly calculated, so it needs to be fixed here
     if "=" in prefix and "=" not in commprefix:
         strip_len = prefix.index("=") + 1
+    # Fix cases where stdout/stderr is being redirected (e.g. 'ls >out.txt')
```

Comment:
instead of doing this by hand, have you tried using the xonsh parser or perhaps update completion parser ?

will this approach work when the arg has > inbetween ?

Reply:
My current logic should work when '>' is in the middle, but I like your idea of using the parser better. I will look into that.